### ISSUE TITLE
object-shorthand rule incorrectly detects string values

### ISSUE THREAD

**Issue by C-Higgins** [eslint/eslint#9967]
**Tell us about your environment**

* **ESLint Version:** 4.17.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

When set to "consistent-as-needed", rule detects that the object can be made shorthand, when it cannot.

```js
{
  a: 'a',
  b: 'b',
}
```

**What did you expect to happen?**
No warning

**What actually happened? Please include the actual, raw output from ESLint.**
warning  Expected shorthand for all properties    object-shorthand

**Comment by platinumazure** [comment]
For what it's worth, I can't reproduce using the [demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IG9iamVjdC1zaG9ydGhhbmQ6ZXJyb3IgKi9cclxuKHtcclxuICBhOiAnYScsXHJcbiAgYjogJ2InLFxyXG59KTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjgsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=).

**Comment by C-Higgins** [comment]
Yes, my apologies. A silly unrelated error caused this.
